tests/functional/test_utils.py

```diff
@@ -40,3 +45,20 @@ def test_can_load_token(self):
         loader = FileWebIdentityTokenLoader(self.token_file)
         token = loader()
         self.assertEqual(self.token, token)
+
+
+class TestInstanceMetadataFetcher(BaseEnvVar):
+    def test_catch_retryable_http_errors(self):
+        with mock.patch('botocore.httpsession.URLLib3Session.send') as send_mock:
+            fetcher = InstanceMetadataFetcher()
+            send_mock.side_effect = EndpointConnectionError(endpoint_url="foo")
+            fetcher.retrieve_iam_role_credentials()
+        self.assertEquals(send_mock.call_count, 2)
```

Comment:
Do we have any way to assert the second call is a retry? Maybe pulling a url off `call_args`, `call_args_list`, or `mock_calls`?